interval-timer.js

```diff
@@ -0,0 +1,88 @@
+import { Timer } from "./timer.js";
+import { is_function } from "./util/type-check.js";
+
+class IntervalTime extends Timer {
+  _completed_intervals = [];
+  last_completed_elapsed_time = 0;
+  constructor(
+    on_complete_interval,
+    on_start,
+    on_end,
+    on_update,
+    update_interval_rate
+  ) {
+    super(on_start, on_end, on_update, update_interval_rate);
+    this.on_complete_interval = on_complete_interval;
+  }
+
+  get completed_intervals() {
+    return this._completed_intervals;
+  }
+
+  set completed_intervals(value) {
+    this._completed_intervals.push(value);
+  }
+
+  get completed_interval() {
+    if(!this.elapsed_time){
+      throw Error("Start Timer First");
+    }
+    return this.elapsed_time - this.last_completed_elapsed_time;
+  }
+
+  update_completed_intervals() {
+    this.completed_intervals = this.completed_interval;
+  }
+
+  update_last_completed_elapsed_time() {
+    this.last_completed_elapsed_time = this.elapsed_time;
+  }
+
+  invoke_on_complete_interval() {
+    if (is_function(this.on_complete_interval)) {
+      this.on_complete_interval();
+    }
+  }
+  
+  complete_interval() {
+    this.update_completed_intervals();
+    this.update_last_completed_elapsed_time();
+    this.invoke_on_complete_interval();
+  }
+
+  retrieve_all_completed_intervals() {
+    this.completed_intervals.forEach((e, i) =>
```

Comment:
Let's make this return the completed intervals array instead of console.log

Reply:
Resolved it by making function retrieve_all_completed_intervals() to return the completed intervals array